frontend/src/index.tsx

```diff
@@ -21,22 +22,39 @@ function getInitialScheme(): 'light' | 'dark' {
   }
 }
 
+posthog.init(import.meta.env.VITE_PUBLIC_POSTHOG_KEY, {
+  api_host: import.meta.env.VITE_PUBLIC_POSTHOG_HOST,
+  defaults: '2025-05-24',
+  capture_exceptions: true, // This enables capturing exceptions using Error Tracking, set to false if you don't want this
+  debug: false,
+  opt_out_capturing_by_default: false, // We handle opt-out via cookie consent
+});
+
+function UpdatePosthogConsent(){
+    if(typeof(posthog) == "undefined") {
+      return;
+    }
+    const optIn = (window.CookieConsent as any).acceptedCategory('analytics');
+    optIn?
+      posthog.opt_in_capturing() : posthog.opt_out_capturing();
+
+    console.log("Updated analytics consent: ", optIn? "opted in" : "opted out");
+  }
+
+window.addEventListener("cc:onConsent", UpdatePosthogConsent);
+window.addEventListener("cc:onChange", UpdatePosthogConsent);
```

Comment:
```suggestion
function updatePosthogConsent(){
    if(typeof(posthog) == "undefined") {
      return;
    }
    const optIn = (window.CookieConsent as any).acceptedCategory('analytics');
    optIn?
      posthog.opt_in_capturing() : posthog.opt_out_capturing();

    console.log("Updated analytics consent: ", optIn? "opted in" : "opted out");
  }

window.addEventListener("cc:onConsent", updatePosthogConsent);
window.addEventListener("cc:onChange", updatePosthogConsent);
```

Should really be lowerCamelCase